File: ruby/resistor-color-duo/resistor_color_duo.rb
class ResistorColorDuo
  COLORS = { "black" => 0, "brown" => 1,
             "red" => 2, "orange" => 3,
             "yellow" => 4, "green" => 5,
             "blue" => 6, "violet" => 7,
             "grey" => 8, "white" => 9
           }
  def self.value(colors)
    colors.first(2).map { |color| COLORS[color] }.join.to_i
  end
end

File: javascript/resistor-color-duo/resistor-color-duo.js
// value : array of 2 strings
const COLORS = ['black', 'brown', 'red', 'orange', 'yellow', 'green', 'blue', 'violet', 'grey', 'white'];
const colorCode = (color) => COLORS.indexOf(color);
export const decodedValue = (value) => {
  return (colorCode(value[0]) * 10 + colorCode(value[1]));
};
